### PR TITLE
Modify configure.py to meet some distibution requirements (build separated python and C++ libraries)

### PR DESCRIPTION
**Pull request by ghost**
This can help to create packages for binary distributions

### CODE REVIEW

File: configure.py [hunk 1]
```diff
@@ -58,38 +59,50 @@ def __init__(self, qmake):
...
     )
+    parser.add_argument(
+        '--lib-path',
+        dest="lib_path",
+        type=str,
+        default="",
+        help="Add path to the built QOnScreenKeyboard library. This will disable building it",
+    )
     args=parser.parse_args()
+    if args.lib_path != "":
```
Comment (spacok) [inline configure.py]: It's string right? so we can use "not args.lib_path" here.

Comment (takada07) [inline configure.py]: If this is string, you can use "not" here. For making code more readable....

File: configure.py [hunk 2]
```diff
@@ -58,38 +59,50 @@ def __init__(self, qmake):
...
+                print(
+                    "Use the --qmake argument to explicitly specify a "
+                    "working Qt qmake."
+                )
+            exit(1)
 
     sip_args=args.sip_extras
-
+    rundir=os.path.dirname(sys.argv[0])
+    if rundir == "":
```
Comment (takada07) [inline configure.py]: same as line 70

File: configure.py [hunk 3]
```diff
@@ -43,7 +44,7 @@ def __init__(self, qmake):
 if __name__=="__main__":
     from argparse import ArgumentParser
 
-    parser=ArgumentParser(description="Configure PyMyLabel module.")
+    parser=ArgumentParser(description="Configure QOnScreenKeyboard library.")
```
Comment (takada07) [inline configure.py:47]: heh, thanks, this is there because of sip and pyqt5 tutorial :)

File: configure.py [hunk 4]
```diff
@@ -58,38 +59,50 @@ def __init__(self, qmake):
...
     pyconfig=HostPythonConfiguration()
     py_sip_dir=os.path.join(pyconfig.data_dir, 'sip', 'PyQt5')
     sip_inc_dir=pyconfig.venv_inc_dir
 
     qtconfig=TargetQtConfiguration(qmake_exe)
 
-    inc_dir=os.path.abspath(os.path.join(".","src"))
-    lib_dir=inc_dir
+    inc_dir=os.path.abspath(os.path.join(rundir,"src"))
+    if args.lib_path == "":
```
Comment (takada07) [inline configure.py]: again... line 70

File: configure.py [hunk 5]
```diff
@@ -149,20 +162,23 @@ def __init__(self, qmake):
...
-        subdirs = ["src", output_dir],
-    ).generate()
-
-    os.chdir("src")    
-    qmake_cmd=qmake_exe
-    if sys.platform=="win32": qmake_cmd+=" -spec win32-msvc2010"
-    print()
-    print(qmake_cmd)
-    os.system(qmake_cmd)
+    if args.lib_path == "":
```
Comment (takada07) [inline configure.py]: line 70

**Comment by takada07** [comment]
OK, if I understand it correctly, this can produce separatet python module. It will disable build of libQOnScreenKeyboard.so. What if I want to build only separated library without python bindings?

**Comment by ghost** [comment]
Yes I was thinking little bit about that. But basically you can run just qmake on .pro in src directory, you don't need to run configure.py. This is the easiest way. But it can be somehow obfuscated in configure to generate top level makefile which will only call make in src directory. Not a problem but I think it is not necessary now. Code is easy so this is simple enough

**Comment by takada07** [comment]
Looks good.

**Pull request merged**